### PR TITLE
GitHub: Auto format XAML command

### LINKED ISSUE #13396: Code Quality: Update the XAML styler check workflow to apply styling as well
### Description

As the title.

### Concerned code

The workflow that runs XAML styler console checker.

### Gains

- Better maintainability.

### Requirements

- Update the file to do so

### Comments

https://github.com/CommunityToolkit/Labs-Windows/blob/main/ApplyXamlStyling.ps1

### PR DESCRIPTION
**Pull request by heftymouse**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #13396

**Validation**
How did you test these changes?
- [ ] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   1. Open app ...
   2. Click settings button ...

**Screenshots (optional)**
Add screenshots here.

PRs with XAML files will be formatted when a comment consisting of `/format` is created and 'allow edits by maintainers' is enabled. Requires a bot account to be created with its token added as a secret `FILES_BOT_TOKEN`.

### CODE REVIEW

**Review by 0x5bfa — APPROVED** [review]
LGTM in terms of code quality.

I feel this kind of change is hard to be tested. I suggest to add a XAML file formatted wrongly, so we can test easily.

And, if you are inclined to use Bots, we are going to need to create a GitHub Apps inside this organization and install it to this organization. And due to its nature, GitHub Apps doesn't provide Access Token, therefore, you may be going to need to add following task if I didn't misunderstand the context.

```yml
- name: Generate GitHub Apps token
  id: generate
  uses: tibdex/github-app-token@v1
  with:
    app_id: ${{ secrets.APP_ID }}
    private_key: ${{ secrets.PRIVATE_KEY }}
```

**Comment by yaira2** [comment]
@heftymouse can we reuse the token from the GitHub integration, or is this something else?

**Comment by heftymouse** [comment]
> @heftymouse can we reuse the token from the GitHub integration, or is this something else?

Technically yes, but it's better to create a separate GitHub app since the client secret for the GitHub integration in the app is public

**Comment by yaira2** [comment]
> the client secret for the GitHub integration in the app is public

It is? I thought we inject that through the pipeline.

**Comment by heftymouse** [comment]
It'll be visible in the decompiled app

**Comment by yaira2** [comment]
We can create a separate one for this 👍

**Comment by Jay-o-Way** [comment]
@crutkas would this (apply versus check) be something for PowerToys repo to?

**Comment by 0x5bfa** [comment]
@Jay-o-Way  Yes.

@heftymouse We need to use app id and secret but workflow triggered by pull_request event cannot access to our secrets. We need it to be *pull_request_target* and then check & update xaml formatting. Due to its nature, we cannot test this workflow until this PR will have merged.

**Comment by heftymouse** [comment]
@0x5bfa This action uses issue_comment, not pull_request

**Comment by crutkas** [comment]
Maybe. Would want to sync with people, could also be a daily cron job. Would love to see one of the PRs it creates.

**Comment by 0x5bfa** [comment]
@heftymouse, my bad. you can use secrets from workflow triggered by issue_comment. FYI, workflow triggered by pull_request, pull_request_review, pull_request_review_comment, and push cannot access to the secrets.

**Comment by heftymouse** [comment]
> Would love to see one of the PRs it creates.
@crutkas Here's one: https://github.com/heftymouse/Files/pull/8/commits/530d9fc099e605143fef7166a164e652d80c4c31

File: .github/workflows/format-xaml.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    defaults:
+      run:
+        shell: pwsh
+
+    steps:
+      - name: Generate GitHub Apps token
+        id: generate
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{ secrets.FILES_APP_ID }}
```
Comment (yaira2) [inline .github/workflows/format-xaml.yml:20]:
```suggestion
          app_id: ${{ secrets.BOT_APP_ID }}

```

File: .github/workflows/format-xaml.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+          private_key: ${{ secrets.FILES_PRIVATE_KEY }}
```
Comment (yaira2) [inline .github/workflows/format-xaml.yml:21]:
```suggestion
          private_key: ${{ secrets.BOT_PRIVATE_KEY }}

```

**Comment by yaira2** [comment]
@heftymouse thank you!

**Pull request merged**